### PR TITLE
OCPBUGS-18455: Add CloudControllerManager field for external platform

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
Introduces a new configuration to the platform external platform spec which allows an external cloud provider to be enabled.

Ref: 
https://github.com/openshift/installer/pull/7533
https://github.com/openshift/installer/pull/7457 

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: go.mod
Comment (andfasano) [inline go.mod:1]: Not sure if the github view is correct but.. how did you vendored this file? I was expecting a version bump in go.mod

**Review by zaneb — COMMENTED** [review]
This change will be needed at some point for the benefit of assisted, but for the purposes of fixing the bug it's [not required](https://github.com/openshift/installer/pull/7533/files#r1340758629).

I think you created this PR to avoid a backwards compatibility issue in future if the agent installer doesn't generate this field in its ACI manifests? On reflection, I think that can be solved by assisted-service always setting `CloudControllerManager: External` when the `oci` platform is suggested. I added a [comment on the bug](https://issues.redhat.com/browse/OCPBUGS-18455?focusedId=23172067&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23172067) to that effect.

So at this stage, I think that we can defer this change until there is actually an implementation to put behind this API field.

**Comment by pawanpinjarkar** [comment]
> I think you created this PR to avoid a backwards compatibility issue in future if the agent installer doesn't generate this field in its ACI manifests?

That is right @zaneb

**Comment by pawanpinjarkar** [comment]
The code changes are not required to fix the bug but to address another issue as mentioned by Zane above. Closing as its not needed for the bug.

**Pull request closed without merging**